### PR TITLE
Upgrade dependencies and use config files

### PR DESCRIPTION
**Pull request by hosamaly**
This started as just upgrading dependencies while trying out the SDLC with GitHub, TeamCity, etc. Then I noticed that it depended on the local file system structure, making it impossible to run on TeamCity. I took some time to fix that dependency.

### CODE REVIEW

File: src/main/scala/com/gu/kindlegen/Main.scala [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
+    case Success(settings) => run(settings)
+    case Failure(error) => handleConfigError(error)
+  }
+
+  private def run(settings: Settings): Unit = {
+    val kindleGenerator = new KindleGenerator(settings)
+    kindleGenerator.getNitfBundleToDisk
+    println("Done!")
+    // Why does the program not exit here?
+  }
```
Comment (hosamaly) [inline src/main/scala/com/gu/kindlegen/Main.scala]: Nothing changed in this method; it's the same as the body of this class used to be.

File: src/main/scala/com/gu/kindlegen/Querier.scala [hunk 2]
```diff
@@ -77,12 +64,12 @@ object Querier {
   }
 
   def sortContentByPageAndSection(response: Seq[Content]): Seq[Content] = {
-    response.sortBy(content => (content.fields.flatMap(_.newspaperPageNumber), content.tags.find(_.`type` == NewspaperBookSection).get.id))
+    response.sortBy(content => (content.fields.flatMap(_.newspaperPageNumber), content.tags.find(_.`type` == NewspaperBookSection).map(_.id)))
```
Comment (hosamaly) [inline src/main/scala/com/gu/kindlegen/Querier.scala]: This is a tiny side-fix to avoid issues when there is no content with the expected tag.

Comment (mchv) [reply src/main/scala/com/gu/kindlegen/Querier.scala]: 👍

**Comment by mchv** [comment]
Good work @hosamaly

**Pull request merged**